glossary expanded by soil moisture terms

vgro opened the pull request:
This PR added definitions of different soil moisture related variables to the glossary. Please check if correct and if any important terms are missing.

Fixes #772 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

vgro committed:
glossary expanded by soil moisture terms

vgro committed:
Merge branch 'develop' into 772-add-definitions-of-different-soil-moistures-to-glossary

lelavathy commented on docs/source/glossary.md:
```diff
@@ -33,4 +33,37 @@ LMWC
+  A dimensionless measure comparing current soil moisture to the range between residual
+  and field capacity. Expressed as a unitless ratio (0–1) or %; can also be related to
+  mm if total available water is known.
+
```
please check the definition:
a ratio expressing the amount of water in soil, as a percentage, compared to the total water-holding capacity between the wilting point (very dry) and saturation (very wet) for a given soil type. Expressed as a unitless ratio (0–1) or %.

formula: RSM=soil moisture-wilting point/saturation-wilting point

references:
1. https://weather.ndc.nasa.gov/sport/training/articles/20170320184849_NASA-Land-Information-System-LIS-Soil-Moisture-Percentile/media/lis-soil-moisture-quick-guide-by-nasa-sport.pdf#:~:text=Relative%20soil%20moisture%20is%20a%20ratio%20of,and%20is%20also%20expressed%20as%20a%20percentage.&text=0%2D10%20cm%20Relative%20Soil%20Moisture%20(RSM)%20(%)%2C,saturation%20points%20for%20a%20given%20soil%20type.
2. https://crops.extension.iastate.edu/post/facts-soil-moisture-benchmarking-tool

lelavathy commented on docs/source/glossary.md:
```diff
@@ -33,4 +33,37 @@ LMWC
+  layer, or (less commonly) the derivative of the water retention curve (specific
+  moisture capacity). Alternative meaning (specific moisture capacity): 1/m or unitless
+  per unit suction (if used in hydrological modeling)
+
```
missing soil moisture terms:
1.permanent wilting point: The water content at which plants can no longer extract moisture, leading to wilting and potential plant death. Water is still present in the soil, but it is tightly bound and unavailable to plants. This is lower limit of plant-available water. Expressed in m3/m3, %, or mm                                                                                                                                                                        

2.  available water capacity: amount of water that a soil can store and make available to plants, defined as the difference between field capacity and the permanent wilting point. Expressed in m3/m3, %, or mm 

3. saturation soil moisture: maximum amount of water a soil can hold when all its pores are completely filled with water — that is, the soil is fully saturated and contains no air in the pore spaces. Expressed in m3/m3, %, or mm.

reference: 
1.https://www.fao.org/4/y4690e/y4690e04.htm
2.https://www.droughtmanagement.info/literature/BC_MA_Soil_Water_Storage_Capacity_2005.pdf
3.Rai, R. K., Singh, V. P., & Upadhyay, A. (2017). Planning and evaluation of irrigation projects: methods and implementation. Academic press.
4.https://connectedcrops.ca/the-ultimate-guide-to-soil-moisture/

vgro committed:
added missing soil moisture terms

vgro commented:
@lelavathy I have updated the relative soil moisture and added the missing terms. Could you please check the changes and approve when you're happy with it :-)

vgro committed:
Merge branch 'develop' into 772-add-definitions-of-different-soil-moistures-to-glossary
